assets/wizards/popups/views/segmentation/SingleSegment.js

```diff
@@ -0,0 +1,114 @@
+/**
+ * WordPress dependencies.
+ */
+import { useEffect, useState } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { find } from 'lodash';
+
+/**
+ * Internal dependencies.
+ */
+import { Button, TextControl, CheckboxControl, Router } from '../../../../components/src';
+
+const { NavLink, useHistory } = Router;
+
+const SegmentSettingSection = ( { title, children } ) => (
+	<div className="newspack-campaigns-wizard-segments__section">
+		<h3>{ title }</h3>
+		<div className="newspack-campaigns-wizard-segments__section__content">{ children }</div>
+	</div>
+);
+
+const SegmentsList = ( { segmentId, wizardApiFetch } ) => {
+	const [ name, setName ] = useState( '' );
+	const [ min_posts, setMinPosts ] = useState( 0 );
+	const [ max_posts, setMaxPosts ] = useState( 0 );
+	const history = useHistory();
+
+	const isSegmentValid = name.length > 0;
+
+	const isNew = segmentId === 'new';
+	useEffect( () => {
+		if ( ! isNew ) {
+			wizardApiFetch( {
+				path: `/newspack/v1/wizard/newspack-popups-wizard/segmentation`,
+			} ).then( segments => {
+				const foundSegment = find( segments, ( { id } ) => id === segmentId );
+				if ( foundSegment ) {
+					setName( foundSegment.name );
+					setMinPosts( foundSegment.configuration.min_posts );
+					setMaxPosts( foundSegment.configuration.max_posts );
+				}
+			} );
+		}
+	}, [ isNew ] );
+
+	const saveSegment = () => {
```

Comment:
@adekbadek do you think we need some light validation here to ensure that `max_posts` is greater than `min_posts` if both are enabled?

Reply:
I see the "save" button is disabled in this case—thanks!

Reply:
Whoops, unresolving because I was wrong (was just missing a title)

Reply:
(responded in [review comment](https://github.com/Automattic/newspack-plugin/pull/689#issuecomment-720647687))